.github/workflows/internal-ci-checks.yml

```
@@ -0,0 +1,28 @@
+name: Internal CI Checks
+# This workflows checks if the author of a PR is a member of the CodeQL team
+# and adds `ready-for-internal-ci` label to trigger the internal CI
+
+on:
+  pull_request_target:
+
+jobs:
+  set-label:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set a label to trigger internal CI checks
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          USERNAME="${{ github.event.pull_request.user.login }}"
+          LABEL="ready-for-internal-ci"
```

It's unlikely that `pull_request.user.login` contains dangerous symbols like `"` and `$()` but better safe than sorry:
```suggestion
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          USERNAME: ${{ github.event.pull_request.user.login }}
          PR_NUMBER: ${{ github.event.pull_request.number }}
          LABEL: "ready-for-internal-ci"
        run: |
```